### PR TITLE
drivers: dma for stm32u5 returns from a repeated start or stop command

### LINKED ISSUE #54414: stm32u5 dma driver does not support repeated start-stop
**Describe the bug**
When running the tests/drivers/dma/loop_transfer/  test case on a stm32u5x board, the test_dma_m2m_loop_repeated_start_stop  fails.               

--> on nucleo_u575zi_q or b_u585i_iot02a target board

The DMA driver for the stm32u5 should return correctly on a repeated start or stop command.

**To Reproduce**
1. west build -p auto -b nucleo_u575zi_q tests/drivers/dma/loop_transfer
2. west flash
3. See error

**Expected behavior**
Test PASSED

**Logs and console output**
```
START - test_dma_m2m_loop_repeated_start_stop                                   
DMA memory to memory transfer started                                           
Preparing DMA Controller                                                        
this platform do not support the dma channel                                    
Starting the transfer on channel 8 and waiting for 1 second                     
ERROR: unfinished transfer                                                      
                                                                                
    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/dma/loop_transfer/src/e
                                                                                
 FAIL - test_dma_m2m_loop_repeated_start_stop in 0.286 seconds                  

```

**Environment (please complete the following information):**
 - OS: ubuntu
 - Toolchain  Zephyr SDK 0.15.0
 - Commit SHA  fcd60cf789b1acd1cd29dcd8db2dba86572bdbfa
..

### PR DESCRIPTION
**Pull request by FRASTM**
On a repeated dma_start, the stm3u5 dma driver should return immediately if it is already started.
On a repeated dma_stop, the stm3u5 dma driver should return immediately if it is already stopped.
This like the dma driver does for other stm32 series.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/54414

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
$ west build -p auto -b nucleo_u575zi_q tests/drivers/dma/loop_transfer/D

```
===================================================================             
TESTSUITE dma_m2m_loop succeeded                                                
                                                                                
------ TESTSUITE SUMMARY START ------                                           
                                                                                
SUITE PASS - 100.00% [dma_m2m_loop]: pass = 3, fail = 0, skip = 0, total = 3 dus
 - PASS - [dma_m2m_loop.test_dma_m2m_loop] duration = 0.605 seconds             
 - PASS - [dma_m2m_loop.test_dma_m2m_loop_repeated_start_stop] duration = 0.605s
 - PASS - [dma_m2m_loop.test_dma_m2m_loop_suspend_resume] duration = 0.866 secos
                                                                                
------ TESTSUITE SUMMARY END ------                                             
                                                                                
===================================================================             
PROJECT EXECUTION SUCCESSFUL                                                    

```

File: drivers/dma/dma_stm32u5.c [hunk 1]
```diff
@@ -188,6 +188,14 @@ void stm32_dma_enable_stream(DMA_TypeDef *dma, uint32_t id)
...
 
+bool stm32_dma_is_enabled_stream(DMA_TypeDef *dma, uint32_t id)
+{
+	if (LL_DMA_IsEnabledChannel(dma, dma_stm32_id_to_stream(id)) == 1) {
+		return true;
+	}
+	return false;
+}
+
 int stm32_dma_disable_stream(DMA_TypeDef *dma, uint32_t id)
```
Comment (gautierg-st) [inline drivers/dma/dma_stm32u5.c:199]: In this function you have a call to LL_DMA_IsEnabledChannel (line 207), that could be changed to use your new function.
Sorry for not putting this comment under the specific line directly, but apparently Github does not allow leaving a comment to a line too far from a modification...

**Review by gautierg-st — DISMISSED** [review]

File: drivers/dma/dma_stm32u5.c (hunk 1 above)
Comment (FRASTM) [reply drivers/dma/dma_stm32u5.c:199]: true, but...
it keeps the driver like it is done in the dma_stm32_v1 or dma_stm32_v2  and that will probably save few cpu instructions at runtime.

Comment (FRASTM) [reply drivers/dma/dma_stm32u5.c:199]: @erwango should the driver stm32_dma_disable_stream call the stm32_dma_is_enabled_stream
here and also in _dma_stm32_v1.c_ or _dma_stm32_v2.c_ ?

Comment (erwango) [reply drivers/dma/dma_stm32u5.c:199]: I think we should try to make it simple.
Dedicated U5 driver was done in order to simplify things. I don't think we should add complexity to get all drivers look similar.

**Comment by FRASTM** [comment]
Change to use the new stm32_dma_is_enabled_stream() function inside the stm32_dma_disable_stream() (only for the driver of this stm32u5 serie).

**Review by erwango — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**